In today’s rapidly changing environment, climate change is no longer just an ecological concern—it is a critical business and economic challenge. Companies, policymakers, and investors need data-driven insights to assess climate risks, understand economic vulnerabilities, and develop strategies for sustainability and resilience.



Assess Climate Vulnerability: Identify how climate-related events impact health, economic growth e.t.c

Quantify Financial Risks: Measure economic losses due to climate-related disasters and forecast future risks.

Develop Risk Mitigation Strategies: Provide actionable insights to improve climate resilience and sustainability.

Enable Data-Driven Decision-Making: Utilize AI-powered models to predict the long-term effects of climate change.

Using the Climate Risk Index dataset, we provide comprehensive analysis on climate change’s impact on public health, economic stability, and global risk factors.

In [2]:
import pandas as pd



In [5]:
# Load and read the CSV
climate_risk_df = pd.read_csv("Data/Climate Change - datasets/Climate Risk Index.csv")


In [6]:
# Display the first few rows to understand its structure
climate_risk_df.head()

index  cartodb_id  the_geom  the_geom_webmercator       country  cri_rank  \
0      0           1       NaN                   NaN  Saudi Arabia        79   
1      1           2       NaN                   NaN       Romania        61   
2      2           3       NaN                   NaN         Spain        69   
3      3           4       NaN                   NaN      Slovenia       135   
4      4           5       NaN                   NaN   South Sudan       133   

   cri_score  fatalities_per_100k_rank  fatalities_per_100k_total  \
0      72.50                        18                       0.45   
1      61.50                       112                       0.01   
2      66.33                        74                       0.05   
3     124.50                       114                       0.00   
4     117.33                       114                       0.00   

   fatalities_rank  fatalities_total  losses_per_gdp__rank  \
0               18               140                   131   
1              102                 1                    16   
2               47                22                    86   
3              114                 0                   135   
4              114                 0                   120   

   losses_per_gdp__total  losses_usdm_ppp_rank  losses_usdm_ppp_total  \
0                 0.0001                   119                  1.229   
1                 0.6746                    11               2797.884   
2                 0.0394                    31                637.070   
3                    NaN                   135                  0.000   
4                 0.0021                   122                  0.508   

  rw_country_code rw_country_name  
0             SAU    Saudi Arabia  
1             ROU         Romania  
2             ESP           Spain  
3             SVN        Slovenia  
4             SSD     South Sudan

In [7]:
climate_risk_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 182 entries, 0 to 181
Data columns (total 17 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   index                      182 non-null    int64  
 1   cartodb_id                 182 non-null    int64  
 2   the_geom                   0 non-null      float64
 3   the_geom_webmercator       0 non-null      float64
 4   country                    182 non-null    object 
 5   cri_rank                   182 non-null    int64  
 6   cri_score                  182 non-null    float64
 7   fatalities_per_100k_rank   182 non-null    int64  
 8   fatalities_per_100k_total  182 non-null    float64
 9   fatalities_rank            182 non-null    int64  
 10  fatalities_total           182 non-null    int64  
 11  losses_per_gdp__rank       182 non-null    int64  
 12  losses_per_gdp__total      131 non-null    float64
 13  losses_usdm_ppp_rank       182 non-null    int64  

The dataset consists of 182 entries and 17 columns, with key information on climate risk, fatalities, and economic losses. Here’s a breakdown:

Key Observations:

1. Climate Risk Index (CRI)

cri_rank: Ranking of countries based on climate risk (lower = more vulnerable).

cri_score: Overall climate risk score (higher = more risk).

2. Health Impact (Fatalities)

fatalities_per_100k_total: Climate-related fatalities per 100,000 people.

fatalities_total: Total fatalities from climate-related events.

fatalities_rank: Ranking of countries by climate-related deaths.

3. Economic Impact

losses_per_gdp__total: Percentage of GDP lost due to climate disasters (131 non-null values, meaning some missing data).

losses_usdm_ppp_total: Total financial losses in million USD (adjusted for purchasing power parity).

losses_per_gdp__rank: Rank based on GDP losses.

4. Country Information

country: Name of the country.

rw_country_code and rw_country_name: Country identifiers (some missing values).

5. Missing Data

the_geom and the_geom_webmercator are empty (not useful for analysis).

losses_per_gdp__total has 51 missing values.
